### ISSUE TITLE
some suggestion in support web css

### ISSUE THREAD

**Issue by PeterChen1997** [redwoodjs/graphql#5415]
I think maybe we can add some support for css

such as
- add support for css modules in ssr build process (shows styles undefined)
- add support for css modules in ts
- generate xxx.module.scss file automatically (support config generate or not and generate type(scss\less\stylus...))
- support config postcss(such as auto import some mixin or functions)
- ...

<img width="833" alt="image" src="https://user-images.githubusercontent.com/21072589/166619617-3eabf07f-a1c6-4268-9a2e-47fc9b8f40a5.png">

**Comment by jtoar** [comment]
@PeterChen1997 agreed, we could do a lot more, so happy to dicusss! Which of those features you enumerated do you use or would like to see?

Note that we do support configuring postcss. If you make a `web/config/postcss.config.js` file, Redwood's build process will pick it up and merge it in. That's how Tailwind CSS currently works with Redwood.

**Comment by PeterChen1997** [comment]
got it, I will have a look, thx!

**Comment by minemax-ua** [comment]
Redwoodjs claims to be a full-stack app framework, but, speaking of styling, out of the box, it supports what... Nothing but "good old" css files? Really?

**Comment by dthyresson** [comment]
> Redwoodjs claims to be a full-stack app framework, but, speaking of styling, out of the box, it supports what... Nothing but "good old" css files? Really?

Hi @minemax-ua.

The RedwoodJS `setup ui` [command](https://redwoodjs.com/docs/cli-commands#setup-ui)

> Set up a UI design or style library. Right now the choices are [Chakra UI](https://chakra-ui.com/), [TailwindCSS](https://tailwindcss.com/) and [WindiCSS](https://windicss.org/).

Mantine support is coming soon, too. And Chakra 2 soon when React 18 support is in.

Picking a design/styling tool, I find, is very personal and designers have their preferences. Some like Styled Components, some like TailwindCSS, some like to handcode their own.

So, while out-of-the-box, RedwoodJS has chosen to go minimalist (css) because most things build from there.

It's rather common for the next step after creating an app, so people to setup a ui option and auth option.

Hope that helps with some context.

**Comment by minemax-ua** [comment]
> > Redwoodjs claims to be a full-stack app framework, but, speaking of styling, out of the box, it supports what... Nothing but "good old" css files? Really?
> 
> Hi @minemax-ua.
> 
> The RedwoodJS `setup ui` [command](https://redwoodjs.com/docs/cli-commands#setup-ui)
> 
> > Set up a UI design or style library. Right now the choices are [Chakra UI](https://chakra-ui.com/), [TailwindCSS](https://tailwindcss.com/) and [WindiCSS](https://windicss.org/).
> 
> Mantine support is coming soon, too. And Chakra 2 soon when React 18 support is in.
> 
> Picking a design/styling tool, I find, is very personal and designers have their preferences. Some like Styled Components, some like TailwindCSS, some like to handcode their own.
> 
> So, while out-of-the-box, RedwoodJS has chosen to go minimalist (css) because most things build from there.
> 
> It's rather common for the next step after creating an app, so people to setup a ui option and auth option.
> 
> Hope that helps with some context.

How can I use CSS Modules?

**Comment by dthyresson** [comment]
> How can I use CSS Modules?

Could you clarify and elaborate on your question?

For example do you mean how might you use https://github.com/css-modules/css-modules ?

Since the RedwoodJS web side is a fundamentally a React app, most ways you'd use other libraries with React typically work ... or are worth trying out first and the seeing what might be needed to smooth out support or improve the DX.

How you would use css modules in another React app and could you share some examples?

**Comment by minemax-ua** [comment]
> > How can I use CSS Modules?
> 
> Could you clarify and elaborate on your question?
> 
> For example do you mean how might you use https://github.com/css-modules/css-modules ?
> 
> Since the RedwoodJS web side is a fundamentally a React app, most ways you'd use other libraries with React typically work ... or are worth trying out first and the seeing what might be needed to smooth out support or improve the DX.
> 
> How you would use css modules in another React app and could you share some examples?

Yes, those CSS Modules.

No, it's not about React, but about the module bundler. And webpack (on which the Redwood is based) does not support CSS Modules out of the box. So, do I have to find and hack it's config files to be able to use css modules?

**Comment by dthyresson** [comment]
> No, it's not about React, but about the module bundler. And webpack (on which the Redwood is based) does not support CSS Modules out of the box. So, do I have to find and hack it's config files to be able to use css modules?

Have you tried
